test/src/test/java/hudson/cli/CLITest.java

```diff
@@ -149,4 +174,95 @@ private void doInterrupt(File jar, File home, FreeStyleProject p, String... mode
         r.waitForCompletion(p.getLastBuild());
     }
 
+    @Test
+    public void reportNotJenkins() throws Exception {
+        home = tempHome();
+        grabCliJar();
+
+        for (String transport: Arrays.asList("-remoting", "-http", "-ssh")) {
+
+            String url = "http://jenkins.io";
```

Comment:
Best to avoid trying to make Internet connections during tests—will make them flaky.